kv/kvserver/concurrency: TestLockTableConcurrentRequests failed

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/2fb3e0de-b95c-40f4-be8e-8b57f7cc69b9?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [2b5baff23f9b53da26f0fc1a763ca43384b464b0](https://github.com/cockroachdb/cockroach/commits/2b5baff23f9b53da26f0fc1a763ca43384b464b0):

Fatal error:

```
panic: test timed out after 53m17s
running tests:
	TestLockTableConcurrentRequests (53m17s)
	TestLockTableConcurrentRequests/concurrency_4 (46m52s)

```
Stack: 

```
goroutine 20643 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests427444142
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1626: numRequests: 10000; numActiveTxns: 4; probability(txn-al reqs): 0.75; probability(new txns): 0.10; probability(duplicate access): 0.50
    lock_table_test.go:1751: txns creted: 691
=== RUN   TestLockTableConcurrentRequests/concurrency_1
items: 10691, aborted: 0, concurrency violations: 9792, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_4
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38242

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/625bcdef-a7a0-45ca-a11c-dc9e31e2891a?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [6e23d32dd0d36181f5c886c9a0eda9c756005bd6](https://github.com/cockroachdb/cockroach/commits/6e23d32dd0d36181f5c886c9a0eda9c756005bd6):

Fatal error:

```
panic: test timed out after 53m17s
running tests:
	TestLockTableConcurrentRequests (53m17s)
	TestLockTableConcurrentRequests/concurrency_128 (7m13s)

```
Stack: 

```
goroutine 55802 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests1417921024
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1626: numRequests: 10000; numActiveTxns: 32; probability(txn-al reqs): 0.90; probability(new txns): 0.75; probability(duplicate access): 0.50
    lock_table_test.go:1751: txns creted: 6741
=== RUN   TestLockTableConcurrentRequests/concurrency_128
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123550 kv/kvserver/concurrency: TestLockTableConcurrentRequests failed [C-test-failure O-robot T-kv]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvserver/concurrency/BUILD.bazel
+++ b/pkg/kv/kvserver/concurrency/BUILD.bazel
@@ -81,6 +81,7 @@ go_test(
         "//pkg/testutils",
         "//pkg/testutils/datapathutils",
         "//pkg/testutils/skip",
+        "//pkg/util",
         "//pkg/util/allstacks",
         "//pkg/util/hlc",
         "//pkg/util/leaktest",

--- a/pkg/kv/kvserver/concurrency/lock_table_test.go
+++ b/pkg/kv/kvserver/concurrency/lock_table_test.go
@@ -33,6 +33,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
 	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -1593,9 +1594,9 @@ func TestLockTableConcurrentRequests(t *testing.T) {
 	probDupAccessWithWeakerStr := 0.5
 	probOnlyRead := possibleProbOnlyRead[rng.Intn(len(possibleProbOnlyRead))]
 
-	if syncutil.DeadlockEnabled {
-		// We've seen 10,000 requests to be too much when running a deadlock build.
-		// Override numRequests to the lowest option (1,000) for deadlock builds.
+	if syncutil.DeadlockEnabled || util.RaceEnabled {
+		// We've seen 10,000 requests to be too much when running a deadlock/race
+		// build. Override numRequests to the lowest option (1,000) for such builds.
 		numRequests = 1000
 	}
 

